sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -727,6 +727,19 @@ object FoldablePropagation extends Rule[LogicalPlan] {
         case j @ Join(_, _, LeftOuter | RightOuter | FullOuter, _) =>
           stop = true
           j
+
+        // Operators that operate on objects should only have expressions from encoders, which
+        // should never have foldable expressions.
+        case o: ObjectConsumer =>
+          stop = true
+          o
+        case o: ObjectProducer =>
+          stop = true
+          o
+        case a: AppendColumns =>
```

Comment:
After think about it more, I think the foldable propagation is still useful for object operators like `DeserializeToObject`, how about we specify the problematic cases here explicitly? i.e. `MapGroups`, `FlatMapGroupsInR`, `CoGroup`